__tests__/app.test.js

```diff
@@ -312,4 +312,117 @@ describe("GET comments by review ID", () => {
+        });
+
+        test("ensure that when extra information is given it is ignored", () => {
+            testComment.extra = "Extra unncessary info";
+            return request(app)
+                .post("/api/reviews/1/comments")
+                .send(testComment)
+                .expect(201)
+                .then(({ body }) => {
+                    const newComment = body.comment;
+                    expect(newComment.author).toEqual(testComment.username);
+                });
+        });
+    });
+
+    describe("sad path :(", () => {
+        test("when no information is sent send back the appropriate error", () => {
+            return request(app)
+                .post("/api/reviews/1/comments")
+                .expect(400)
+                .then(({ body }) => {
+                    expect(body.msg).toEqual("Please provide a username.");
+                });
+        });
+
+        test("when no username is sent send back the appropriate error", () => {
+            return request(app)
+                .post("/api/reviews/1/comments")
+                .send({ body: "test text" })
+                .expect(400)
+                .then(({ body }) => {
+                    expect(body.msg).toEqual("Please provide a username.");
+                });
+        });
+
+        test("when no body is sent send back the appropriate error", () => {
+            return request(app)
+                .post("/api/reviews/1/comments")
+                .send({ username: "username goes here" })
+                .expect(400)
+                .then(({ body }) => {
+                    expect(body.msg).toEqual("Please provide a body.");
+                });
+        });
+
+        test("when an invalid username is sent send back the appropriate error", () => {
+            return request(app)
+                .post("/api/reviews/1/comments")
+                .send({
+                    username: "invalid username",
+                    body: "Oh, here's my head!"
+                })
+                .expect(400)
+                .then(({ body }) => {
+                    expect(body.msg).toEqual("Username provided does not exist.");
+                });
+        });
+
+        test("when an invalid review_id is sent send back the appropriate error", () => {
+          return request(app)
+            .post("/api/reviews/bananas/comments")
+            .send({
+              username: "invalid username",
+              body: "Oh look!",
+            })
+            .expect(400)
+            .then(({ body }) => {
+              expect(body.msg).toEqual("Incorrect format used for query.");
+            });
+        });
+
+        test("when an out of range review_id is sent send back the appropriate error", () => {
+          return request(app)
+            .post("/api/reviews/9001/comments")
+            .send({
+              username: "invalid username",
+              body: "A disembodied voice!",
+            })
+            .expect(400)
```

Comment:
404 not found is more appropriate error code for a valid ID that is not existent in the database

Again I'd make the POST body valid so the test is checking only one error